Signal/src/call/PeerConnectionClient.swift

```diff
@@ -128,17 +144,23 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
     // MARK: - Media Streams
 
     public func createSignalingDataChannel() {
-        let dataChannel = peerConnection.dataChannel(forLabel: Identifiers.dataChannelSignaling.rawValue,
-                                                     configuration: RTCDataChannelConfiguration())
-        dataChannel.delegate = self
+        AssertIsOnMainThread()
+
+        PeerConnectionClient.signalingQueue.sync {
```

Comment:
Should we make this return a promise and refactor the callsite so the signaling queue stuff can run async?